core/src/main/java/org/infinispan/container/offheap/BoundedOffHeapDataContainer.java

```diff
@@ -38,9 +39,13 @@ public BoundedOffHeapDataContainer(int desiredSize, long maxSize, EvictionType t
       this.maxSize = maxSize;
       if (type == EvictionType.COUNT) {
          sizeCalculator = i -> 1;
+         initialSize = 0;
       } else {
          // Use size of entry plus 16 for our LRU pointers
          sizeCalculator = i -> offHeapEntryFactory.getSize(i);
+         // We have to make sure to count the address hash as part of our size
+         initialSize = memoryAddressCount * 8;
```

Comment:
I'd use `<< 3` to match `MemoryAddressHash`

Reply:
Sure.